apis/bigqueryanalyticshub/v1alpha1/bigqueryanalyticshubdataexchange_types.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var BigQueryAnalyticsHubDataExchangeGVK = GroupVersion.WithKind("BigQueryAnalyticsHubDataExchange")
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// BigQueryAnalyticsHubDataExchangeSpec defines the desired state of BigQueryAnalyticsHubDataExchange
+// +kcc:proto=google.cloud.bigquery.analyticshub.v1.DataExchange
+type BigQueryAnalyticsHubDataExchangeSpec struct {
+	// Output only. The resource name of the data exchange.
+	//  e.g. `projects/myproject/locations/US/dataExchanges/123`.
+	Name *string `json:"name,omitempty"`
```

Comment:
Nit: if this is output only, I don't think we want it in spec.  I guess it's in spec because of TF?  If so, let's remove it in a separate PR (we're allowed to do that because alpha)

Reply:
Going to make this change as a separate PR